workers/unity/Packages/com.improbable.gdk.core/Tests/Editmode/Systems/SpatialOSReceiveSystemTests.cs

```diff
@@ -0,0 +1,674 @@
+
+            using (var wrappedOp = WorkerOpFactory.CreateAddEntityOp(TestEntityId))
+            {
+                logDispatcher.Expect(new ExpectedLog(LogType.Error, LoggingUtils.LoggerName, LoggingUtils.EntityId));
+                receiveSystem.OnAddEntity(wrappedOp.Op);
+                logDispatcher.AssertAgainstExpectedLogs();
+            }
+        }
+
+        [Test]
+        public void OnRemoveEntity_should_remove_entity_and_deallocate_world_command_providers()
+        {
+            var entity = SetupTestEntity();
+
+            var createEntityRequestsHandle =
+                entityManager.GetComponentData<WorldCommands.CreateEntity.CommandSender>(entity).Handle;
+            var deleteEntityRequestsHandle =
+                entityManager.GetComponentData<WorldCommands.DeleteEntity.CommandSender>(entity).Handle;
+            var entityQueryRequestsHandle =
+                entityManager.GetComponentData<WorldCommands.EntityQuery.CommandSender>(entity).Handle;
+            var reserveEntityIdsRequestsHandle =
+                entityManager.GetComponentData<WorldCommands.ReserveEntityIds.CommandSender>(entity).Handle;
+
+            using (var wrappedOp = WorkerOpFactory.CreateRemoveEntityOp(TestEntityId))
+            {
+                receiveSystem.OnRemoveEntity(wrappedOp.Op);
+            }
+
+            Assert.IsFalse(entityManager.Exists(entity));
+            Assert.IsFalse(worker.TryGetEntity(new EntityId(TestEntityId), out _));
+
+            Assert.Throws<ArgumentException>(() => WorldCommands.CreateEntity.RequestsProvider.Get(createEntityRequestsHandle));
+            Assert.Throws<ArgumentException>(() => WorldCommands.DeleteEntity.RequestsProvider.Get(deleteEntityRequestsHandle));
+            Assert.Throws<ArgumentException>(() => WorldCommands.EntityQuery.RequestsProvider.Get(entityQueryRequestsHandle));
+            Assert.Throws<ArgumentException>(() => WorldCommands.ReserveEntityIds.RequestsProvider.Get(reserveEntityIdsRequestsHandle));
+        }
+
+        [Test]
+        public void OnRemoveEntity_should_error_if_entity_does_not_exist()
+        {
+            using (var wrappedOp = WorkerOpFactory.CreateRemoveEntityOp(TestEntityId))
+            {
+                logDispatcher.Expect(new ExpectedLog(LogType.Error, LoggingUtils.LoggerName, LoggingUtils.EntityId));
+                receiveSystem.OnRemoveEntity(wrappedOp.Op);
+                logDispatcher.AssertAgainstExpectedLogs();
+            }
+        }
+
+        [Test]
+        public void OnAddComponent_should_be_delegated_to_correct_dispatcher()
+        {
+            using (var wrappedOp = WorkerOpFactory.CreateAddComponentOp(TestEntityId, TestComponentId))
+            {
+                receiveSystem.OnAddComponent(wrappedOp.Op);
+            }
+
+            Assert.IsTrue(componentDispatcher.HasAddComponentReceived);
+        }
+
+        [Test]
+        public void OnAddComponent_should_error_if_unknown_component_id_received()
```

Comment:
Can also test: having two separate dispatchers for two valid component IDs and making sure that receiving an op for one does not go to the other